### PR TITLE
Handle core/edge species/reactions for phase system correctly

### PR DESCRIPTION
**Pull request by hwpang**
### Motivation or Problem
RMG jobs failed when running with pyrms simulators.

### Description of Changes
Ensure species and reactions are handled correctly for phase systems, especially for seed mechanism.

### Testing
Local jobs ran successfully

### Reviewer Tips
Run a job using pyrms simulator with seed mechanism

### CODE REVIEW

**Comment by hwpang** [comment]
New push fixes another bug I found regarding efficiencies. In RMS, the efficiencies dictionary gets rewritten every time we create a phase. If we create the edge phase later, then the efficiencies dictionary contains edge species index, because the edge phase system and core phase system share the core rms reaction objects. If any of the efficiency species are still in edge, the core simulation crashes.

File: rmgpy/rmg/model.py [hunk 1]
```diff
@@ -1443,22 +1443,19 @@ def add_reaction_to_core(self, rxn):
...
+                if not self.core.phase_system.in_nose:
+                    rms_species_list = self.core.phase_system.get_rms_species_list()
+                    species_names = self.core.phase_system.get_species_names()
+                    bits = np.array([spc.molecule[0].contains_surface_site() for spc in rxn.reactants+rxn.products])
+                    if all(bits):
+                        self.core.phase_system.phases["Surface"].add_reaction(rxn,self.edge.phase_system.phases["Surface"])
+                    elif all(bits==False):
+                        self.core.phase_system.phases["Default"].add_reaction(rxn,self.edge.phase_system.phases["Default"])
+                    else:
+                        self.core.phase_system.interfaces[frozenset({"Default","Surface"})].add_reaction(rxn,species_names,rms_species_list)
```
Comment (mjohnson541) [inline rmgpy/rmg/model.py:1458]: I think I get whats going on here, but I found the commit message a bit difficult to understand.

**Review by mjohnson541 — COMMENTED** [review]
Looks pretty good, just one small thing.

**Comment by hwpang** [comment]
New push edits the commit message about modification for `add_reaction_to_core`

**Review by mjohnson541 — APPROVED** [review]

**Comment by amarkpayne** [comment]
@hwpang if you update this branch I'll merge it in. Right now the merge queue is waiting on other PRs to be updated

**Comment by hwpang** [comment]
@amarkpayne It's rebased!

**Review by amarkpayne — APPROVED** [review]
Thanks for the PR @hwpang and thanks for reviewing @mjohnson541 !

**Pull request merged**